synapse/api/errors.py

```diff
@@ -50,12 +51,15 @@ class Codes(object):
 
 
 class CodeMessageException(RuntimeError):
-    """An exception with integer code and message string attributes."""
+    """An exception with integer code and message string attributes.
 
-    def __init__(self, code, msg):
-        super(CodeMessageException, self).__init__("%d: %s" % (code, msg))
+    Attributes:
+        code (int): HTTP error code
+        response_code_message (str): HTTP reason phrase. None for the default.
+    """
+    def __init__(self, code):
+        super(CodeMessageException, self).__init__("%d" % code)
```

Comment:
Right, if this is about sanitising how errors work then I suggest we get rid of `response_code_message` entirely, as it appears to be unused. I do think that we should be passing more than just the code to the base exception though.

> Previously, for an HTTPResponseException, msg was being used for the HTTP reason message, and response_code_message was unused. For a SynapseError, msg was used for the matrix error message, and response_code_message was used for the HTTP reason message.

As far as I see in [errors.py](https://github.com/matrix-org/synapse/blob/f29d85d9e4e0e19ada3b1187439287adc82cfa74/synapse/api/errors.py) `response_code_message` is only ever set by `LimitExceededError`?